### DISCUSSION TITLE
Parquet filter and projection pushdown with Amazon S3 reader

### DISCUSSION THREAD

**Discussion by chrisbrain** [duckdb/duckdb#4559] [General]
Hi - Really liking this tool!

Just a question, would duckdb support the projection and filter pushdowns (e.g. as mentioned here https://duckdb.org/docs/data/parquet) when using the S3 reader (https://duckdb.org/docs/guides/import/s3_import)?

Or with s3 as it's a http transport would it have to load the entire file first and therefore not be able to make use of certain optimisations, e.g. minimising required reads as mentioned on https://duckdb.org/2021/06/25/querying-parquet.html.

**Comment by hannes** [comment]
It does support both, also on the S3 reader.

**Comment by chrisbrain** [comment]
OK thanks. And can you create a view onto an s3 file?

As I understand if you want to do a join between multiple tables which query directly to the parquet file this is the way to achieve this.

**Reply by chrisbrain** [reply]
I think I have answered my own question here, e.g. you can do something like:

```
create view my_view as select * FROM 's3://my-bucket-name/my_data.parquet';
```

And query the parquet directly in s3 - still not sure though if there are more optimisations done of duckdb can read it directly on local disk, e.g. to seek to the required parts in the parquet file to find the relevant data/stats. I assume with s3 it needs to at least get the whole file over http first.

**Reply by Mause** [reply]
> I think I have answered my own question here, e.g. you can do something like:
> 
> ```
> create view my_view as select * FROM 's3://my-bucket-name/my_data.parquet';
> ```
> 
> And query the parquet directly in s3 - still not sure though if there are more optimisations done of duckdb can read it directly on local disk, e.g. to seek to the required parts in the parquet file to find the relevant data/stats. I assume with s3 it needs to at least get the whole file over http first.

We're able to do partial reads via Range requests actually, so it should be fairly efficient
